stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,900 @@
+
+    Returns
+    -------
+    The Jaxns specific log likelihood function.
+
+    """
+    if not jax_avail:
+        raise ImportError("Jax is required")
+
+    if not can_make_gp:
+        raise ImportError("Tinygp is required to make the GP model.")
+
+    @jit
+    def likelihood_model(*args):
+        dict = {}
+        for i, params in enumerate(params_list):
+            if params[0:4] == "log_":
+                dict[params[4:]] = jnp.exp(args[i])
+            else:
+                dict[params] = args[i]
+        kernel = get_kernel(kernel_type=kernel_type, kernel_params=dict)
+        mean = get_mean(mean_type=mean_type, mean_params=dict)
+        gp = GaussianProcess(kernel, times, mean_value=mean(times))
+        return gp.log_probability(counts)
+
+    return likelihood_model
+
+
+class GPResult:
+    """
+    Makes a GPResult object which takes in a Stingray.Lightcurve and samples parameters of a model
+    (Gaussian Process) based on the given prior and log_likelihood function.
+
+    Parameters
+    ----------
+    lc: Stingray.Lightcurve object
+        The lightcurve on which the bayesian inference is to be done
+
+    """
+
+    def __init__(self, lc: Lightcurve) -> None:
+        self.lc = lc
+        self.time = lc.time
+        self.counts = lc.counts
+        self.result = None
+
+    def sample(self, prior_model=None, likelihood_model=None, max_samples=1e4):
+        """
+        Makes a Jaxns nested sampler over the Gaussian Process, given the
+        prior and likelihood model
+
+        Parameters
+        ----------
+        prior_model: jaxns.prior.PriorModelType object
+            A prior generator object.
+            Can be made using the get_prior function or can use your own jaxns
+            compatible prior function.
+
+        likelihood_model: jaxns.types.LikelihoodType object
+            A likelihood fucntion which takes in the arguments of the prior
+            model and returns the loglikelihood of the model.
+            Can be made using the get_log_likelihood function or can use your own
+            log_likelihood function with same order of arguments as the prior_model.
+
+
+        max_samples: int, default 1e4
+            The maximum number of samples to be taken by the nested sampler
+
+        Returns
+        ----------
+        results: jaxns.results.NestedSamplerResults object
+            The results of the nested sampling process
+
+        """
+        if not jax_avail:
+            raise ImportError("Jax is required")
+
+        if not can_sample:
+            raise ImportError("Jaxns not installed! Can't sample!")
+
+        self.prior_model = prior_model
+        self.log_likelihood_model = likelihood_model
+
+        nsmodel = Model(prior_model=self.prior_model, log_likelihood=self.log_likelihood_model)
+        nsmodel.sanity_check(random.PRNGKey(10), S=100)
+
+        self.exact_ns = ExactNestedSampler(nsmodel, num_live_points=500, max_samples=max_samples)
```

Comment:
I think it would be useful to have `num_live_points` as a user-defined parameter. That's something I've seen people want to change.